lib/block.ml

```diff
@@ -145,77 +145,86 @@ let dump ppf ({ loc; section; labels; contents; value; _ } as b) =
     labels
     Fmt.(Dump.option Header.pp)
     (header b)
-    Fmt.(Dump.list dump_string)
+    Fmt.Dump.(list string)
     contents dump_value value
 
-let pp_lines syntax t =
-  let pp =
-    match syntax with
-    | Some Syntax.Cram -> Fmt.fmt "  %s"
-    | Some Syntax.Mli ->
-        fun ppf -> Fmt.fmt "%*s%s" ppf (t.loc.loc_start.pos_cnum + 2) ""
-    | _ -> Fmt.string
-  in
-  Fmt.(list ~sep:(any "\n") pp)
-
-let lstrip strings =
-  let hpad = Misc.hpad_of_lines strings in
-  List.map
-    (fun string ->
-      let first = Util.Int.min (Misc.hpad_of_lines [ string ]) hpad in
-      Astring.String.with_index_range string ~first)
-    strings
-
-let pp_contents ?syntax ppf t =
-  match (syntax, t.contents) with
-  | Some Syntax.Mli, [ line ] -> Fmt.pf ppf "%s" line
-  | Some Syntax.Mli, lines ->
-      Fmt.pf ppf "@\n%a@\n" (pp_lines syntax t) (lstrip lines)
-  | (Some Cram | Some Normal | None), [] -> ()
-  | (Some Cram | Some Normal | None), _ ->
-      Fmt.pf ppf "%a\n" (pp_lines syntax t) t.contents
+let pp_contents ?syntax:_ ppf t =
+  Fmt.(list ~sep:(any "\n") string) ppf t.contents
+
+let rec error_padding = function
+  | [] -> []
+  | [ (`Output _ as o); `Output padding ] when Util.String.all_blank padding ->
+      [ o ]
+  | x :: xs ->
+      let xs = error_padding xs in
+      x :: xs
 
 let pp_errors ppf t =
   match t.value with
-  | OCaml { errors; _ } when List.length errors > 0 ->
-      Fmt.string ppf "```mdx-error\n";
-      Fmt.pf ppf "%a" Fmt.(list ~sep:nop Output.pp) errors;
-      Fmt.string ppf "```\n"
+  | OCaml { errors = []; _ } -> ()
+  | OCaml { errors; _ } ->
+      let errors = error_padding errors in
+      Logs.debug (fun l ->
+          l "Printing errors: %a" (Fmt.Dump.list Output.dump) errors);
+      Fmt.pf ppf "```mdx-error\n%a\n```\n"
+        Fmt.(list ~sep:(any "\n") Output.pp)
+        errors
   | _ -> ()
 
 let pp_footer ?syntax ppf _ =
   match syntax with
-  | Some Syntax.Mli -> ()
-  | Some Syntax.Cram -> ()
-  | _ -> Fmt.string ppf "```\n"
+  | Some Syntax.Mli -> Fmt.string ppf "]}"
+  | Some Syntax.Cram -> Fmt.string ppf "\n"
+  | Some Syntax.Normal | None -> Fmt.string ppf "```\n"
 
 let pp_legacy_labels ppf = function
   | [] -> ()
   | l -> Fmt.pf ppf " %a" Fmt.(list ~sep:(any ",") Label.pp) l
 
-let pp_labels ppf = function
-  | [] -> ()
-  | l -> Fmt.pf ppf "<!-- $MDX %a -->\n" Fmt.(list ~sep:(any ",") Label.pp) l
-
-let pp_header ?syntax ppf t =
+let pp_labels ?syntax ppf labels =
   match syntax with
+  | Some Syntax.Mli -> Fmt.(list ~sep:(any ",") Label.pp) ppf labels
```

Comment:
Agreed. I've added a test for the labels syntax by copying `labels-syntax` and making it an `mli` file. It worked out of the box without issues so that's good.

The disagreements on supported syntax is quite annoying between the various parsers, so over time we should move the `mli` parser closer to the other parsers.